OMCompiler/Compiler/BackEnd/Matching.mo

```diff
@@ -97,9 +97,26 @@ public function RegularMatching "
 algorithm
   ass1 := arrayCreate(nVars, -1);
   ass2 := arrayCreate(nEqns, -1);
-  (ass1, ass2, perfectMatching) := ContinueMatching(m, nVars, nEqns, ass1, ass2, false);
+  (ass1, ass2, perfectMatching, _, _) := ContinueMatching(m, nVars, nEqns, ass1, ass2, false);
 end RegularMatching;
 
+function SingularMatching "
+  Use this function if you expect it to be singular. Also returns marked equations and variables
+  after pantelides algorithm."
+  input BackendDAE.AdjacencyMatrix m;
+  input Integer nVars;
+  input Integer nEqns;
+  output array<Integer> ass1 "eqn := ass1[var]";
+  output array<Integer> ass2 "var := ass2[eqn]";
+  output Boolean perfectMatching;
+  output array<Boolean> eMark;
+  output array<Boolean> vMark;
+algorithm
+  ass1 := arrayCreate(nVars, -1);
+  ass2 := arrayCreate(nEqns, -1);
+  (ass1, ass2, perfectMatching, eMark, vMark) := ContinueMatching(m, nVars, nEqns, ass1, ass2, false);
+end SingularMatching;
```

Comment:
What is the point of this new function? Why not just using (and possibly extending) RegularMatching?

Reply:
I did not want to mess with current structures, but you are probably right.